content/blog/2019-11-21-bug-finding/index.md

```diff
@@ -84,11 +84,11 @@ void safeDereference(int *p){
 }
 ```
 
-However, there is no reliable method to tell an invalid pointer that pointing to a function-scoped variable.
+However, there is no reliable method to identify an invalid pointer that points to a function-scoped variable.
 
 ```c++
 void invalidDereference(int *p){
-	int a = 10;
+    int a = 10;
     if( p != NULL){
     	*pi = a; 
```

Comment:
What is *pi here? Do you mean *p? In any case, I feel like the actual issue is something like this:
```c++
void invalidDereference(int **p){
    int a = 10;
    int a = 10;
    if( p != NULL){
    	*p = &a; 
    }
}
```
Here, we're taking a reference to `a`, which a a function-scoped variable. This is the issue because `a` will be destroyed after the function ends, but we still have a pointer to it, which is bad.